src/Component/JsonSchema/Generator/Normalizer/DenormalizerGenerator.php

```diff
@@ -77,6 +78,20 @@ protected function createDenormalizeMethod(string $modelFqdn, Context $context,
 
         $statements[] = new Stmt\Expression(new Expr\Assign($objectVariable, new Expr\New_(new Name('\\' . $modelFqdn))));
 
+        foreach ($classGuess->getProperties() as $property) {
+            if (Type::TYPE_FLOAT !== $property->getType()->getName()) {
+                continue;
+            }
+            $baseCondition = new Expr\FuncCall(new Name('\array_key_exists'), [
+                new Arg(new Scalar\String_($property->getName())),
+                new Arg($dataVariable),
+            ]);
+            $arrayElement = new Expr\ArrayDimFetch($dataVariable, new Scalar\String_($property->getName()));
+            $intCondition = new Expr\FuncCall(new Name('\is_int'), [$arrayElement]);
```

Comment:
I'm not sure `is_int` would work with float numbers, shouldn't we use `is_numeric` here instead ?

Reply:
i explicitly want to detect when the value is an integer to cast it to float. if it already is a float, we don't need to do anything.

the generated code looks like this:
```
if (\array_key_exists('field', $data) && \is_int($data['field'])) {
    $data['field'] = (double) $data['field'];
}
```

Reply:
Ok I see :+1: Thanks for the fix, could you also add a test on float number in src/Component/JsonSchema/Tests/Validation/ValidationTest.php ?

Reply:
i have this test: https://github.com/janephp/janephp/pull/643/files#diff-2c71a12f47f12f4724d5ffd6f9bbd4cac7b803ee5a49f6f19b4516d87e6e0a0aR712

is this not what you meant?

Reply:
I was meaning with a float number, to check that old behavior is still fine. Like: `7,7`.

Reply:
that test already exists on the lines above the one i added: https://github.com/janephp/janephp/pull/643/files#diff-2c71a12f47f12f4724d5ffd6f9bbd4cac7b803ee5a49f6f19b4516d87e6e0a0aR706

Reply:
my bad, thanks for your work !